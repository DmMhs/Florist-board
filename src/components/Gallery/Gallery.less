.Gallery {
  min-height: calc(100vh - 7rem);
  border: 1px solid transparent;
  box-sizing: border-box;
  h2 {
    color: #333333;
    font-weight: normal;
  }
  hr {
    width: 50%;
    border: 0.5px solid #eee;
  }
  .images-wrapper {
    display: flex;
    width: 80%;
    margin: 0 auto;
    flex-wrap: wrap;
    box-sizing: border-box;
    margin-bottom: 1rem;
    .column {
      -ms-flex: 25%; /* IE10 */
      flex: 25%;
      max-width: 25%;
      img {
        position: relative;
        width: calc(100% - 8px);
        margin-top: 8px;
        vertical-align: middle;
        border: 1px solid #eee;
        &:hover {
          cursor: pointer;
          box-shadow: 1px 1px 3px #ccc;
        }
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .Gallery {
    margin-top: 4rem;
    .images-wrapper {
      .column {
        -ms-flex: 50%; /* IE10 */
        flex: 50%;
        max-width: 50%;
      }
    }
  }
}
@media screen and (max-width: 576px) {
  .Gallery {
    .images-wrapper {
      .column {
        -ms-flex: 100%; /* IE10 */
        flex: 100%;
        max-width: 100%;
      }
    }
  }
}
