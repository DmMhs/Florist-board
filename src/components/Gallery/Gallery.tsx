import React, { Component, ReactElement } from 'react';

import './Gallery.less';
import { galleryImagesRef } from '../../firebase';
import Spinner from '../Spinner/Spinner';

interface GalleryProps {}
interface GalleryState {
  images: string[];
  fetchInProgress: boolean;
}

class Gallery extends Component<GalleryProps, GalleryState> {
  constructor(props: GalleryProps) {
    super(props);
    this.state = {
      images: [],
      fetchInProgress: false
    };
  }
  componentDidMount() {
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-1.jpg?alt=media&token=1e2cc5ea-f02c-44a4-90b7-bec8fa221d2c'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-2.jpg?alt=media&token=dc625aef-cc19-4272-a1b8-0b20ada9aefb'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-3.jpg?alt=media&token=472c8f8e-8d33-47f6-bc65-b67662d88794'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-4.JPG?alt=media&token=8784086f-df18-43d9-b9fb-2c438595e6f2'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-5.jpg?alt=media&token=694eaf38-d128-4937-ba58-4754a8a7787d'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-6.jpg?alt=media&token=9333a131-581b-48df-a727-4cf8185b41f3'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-7.jpg?alt=media&token=51cc4591-743f-4d79-a41d-879a6559b1eb'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-8.jpg?alt=media&token=68d05f19-5521-476a-92fc-b69443c22f34'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-9.JPG?alt=media&token=b3f81c5c-2890-4063-a1ab-34a1719e5385'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-10.jpg?alt=media&token=0c20c087-0cc9-45bd-a3e5-16556ead96f2'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-11.jpg?alt=media&token=7a17d6e1-b2e5-4cbc-9008-d153e2d7d570'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-12.jpg?alt=media&token=8e641ef5-8f3c-493a-b916-e15480f60e46'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-13.JPG?alt=media&token=6c0f5cb3-a92a-459b-b7d6-32aa9b5da6c8'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-14.JPG?alt=media&token=f9ea683f-43e9-4230-83d7-5386d77d23e6'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-15.JPG?alt=media&token=d7469b84-57df-42cb-90e9-e610ddf0e24a'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-16.JPG?alt=media&token=77cd0c2f-f6eb-4669-b556-c962d53a626a'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-17.jpg?alt=media&token=20fbdfec-cf2f-48ef-b695-179e5f686200'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-18.JPG?alt=media&token=628808ab-3987-4e81-aa7e-c081b2d15db3'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-19.jpg?alt=media&token=9a4e8daf-1f8a-4190-a29f-6ac7ff4ec52a'
    // );
    // galleryImages.push(
    //   'https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/gallery-images%2Fimage-20.jpg?alt=media&token=40e5875a-db58-4ae5-9ecd-aef411d34617'
    // );
    this.setState({
      fetchInProgress: true
    });
    galleryImagesRef.on('value', snapshot => {
      const newImages: string[] = [];
      snapshot!.forEach((image: firebase.database.DataSnapshot) => {
        newImages.push(image.val());
      });
      this.setState({
        images: newImages,
        fetchInProgress: false
      });
    });
  }
  render() {
    const imagesList = this.state.images.map(
      (imageUrl: string, index: number) => {
        return <img src={imageUrl} alt={`gallery-img-${index}`} key={index} />;
      }
    );
    return (
      <div className="Gallery">
        <h2>Our Bouquets</h2>
        <hr />
        {this.state.fetchInProgress === true ? (
          <Spinner />
        ) : (
          <div className="images-wrapper">
            <div className="column">
              {imagesList.slice(0, Math.floor(imagesList.length / 4))}
            </div>
            <div className="column">
              {imagesList.slice(
                Math.floor(imagesList.length / 4),
                Math.floor(imagesList.length / 4) * 2
              )}
            </div>
            <div className="column">
              {imagesList.slice(
                Math.floor(imagesList.length / 4) * 2,
                Math.floor(imagesList.length / 4) * 3
              )}
            </div>
            <div className="column">
              {imagesList.slice(
                Math.floor(imagesList.length / 4) * 3,
                imagesList.length + 1
              )}
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default Gallery;
