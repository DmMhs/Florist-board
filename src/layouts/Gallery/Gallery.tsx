import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router';

import { galleryImagesRef } from '../../firebase';
import {Spinner} from '../../components';
import Modal from './Modal/Modal';
import { AppContext } from '../../AppContext';
import './Gallery.less';

interface GalleryState {
  images: string[];
  fetchInProgress: boolean;
  showModal: boolean;
  modalIndex: number;
}

class Gallery extends Component<RouteComponentProps<{}>, GalleryState> {
  constructor(props: RouteComponentProps<{}>) {
    super(props);
    this.state = {
      images: [],
      fetchInProgress: false,
      showModal: false,
      modalIndex: 0
    };
  }

  public componentDidMount() {
    this.setState({
      fetchInProgress: true
    });
    galleryImagesRef.on('value', snapshot => {
      const newImages: string[] = [];
      snapshot!.forEach((image: firebase.database.DataSnapshot) => {
        newImages.push(image.val());
      });
      this.setState({
        images: newImages,
        fetchInProgress: false
      });
    });
  }

  private imageClickedHandler = (index: number) => {
    this.setState({
      showModal: true,
      modalIndex: index
    });
  };

  private prevClickedHandler = () => {
    if (this.state.modalIndex !== 0) {
      this.setState({
        modalIndex: this.state.modalIndex! - 1
      });
    }
  };

  private nextClickedHandler = () => {
    if (this.state.images.length > this.state.modalIndex + 1) {
      this.setState({
        modalIndex: this.state.modalIndex! + 1
      });
    }
  };

  private closeModalClickedHandler = () => {
    this.setState({
      showModal: false,
      modalIndex: 0
    });
  };

  public render() {
    const imagesList = this.state.images.map(
      (imageUrl: string, index: number) => {
        return (
          <img
            src={imageUrl}
            alt={`gallery-img-${index}`}
            key={index}
            onClick={this.imageClickedHandler.bind(this, index)}
            className="image"
          />
        );
      }
    );

    const context = this.context;
    const labels = context.state.labels;
    const lang = context.state.lang;

    return (
      <AppContext.Consumer>
        {value => (
          <div className="Gallery">
            <h2>{labels[lang].pages.gallery.main}</h2>
            <hr />
            {this.state.fetchInProgress === true ? (
              <Spinner />
            ) : (
              <div className="images-wrapper">
                <div className="column">
                  {imagesList.slice(0, Math.floor(imagesList.length / 4))}
                </div>
                <div className="column">
                  {imagesList.slice(
                    Math.floor(imagesList.length / 4),
                    Math.floor(imagesList.length / 4) * 2
                  )}
                </div>
                <div className="column">
                  {imagesList.slice(
                    Math.floor(imagesList.length / 4) * 2,
                    Math.floor(imagesList.length / 4) * 3
                  )}
                </div>
                <div className="column">
                  {imagesList.slice(
                    Math.floor(imagesList.length / 4) * 3,
                    imagesList.length + 1
                  )}
                </div>
              </div>
            )}
            {this.state.showModal === true ? (
              <Modal
                images={imagesList}
                initial={this.state.modalIndex}
                prevClickedHandler={this.prevClickedHandler}
                nextClickedHandler={this.nextClickedHandler}
                closeModal={this.closeModalClickedHandler}
              />
            ) : null}
          </div>
        )}
      </AppContext.Consumer>
    );
  }
}

Gallery.contextType = AppContext;

export default withRouter<RouteComponentProps<{}>>(Gallery);
