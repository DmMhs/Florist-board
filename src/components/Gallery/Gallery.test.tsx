import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import Gallery from './Gallery';
import AppContextProvider from '../../AppContext';

it('matches a snapshot', () => {
  const wrapper = mount(
    <BrowserRouter>
      <AppContextProvider>
        <Gallery.WrappedComponent />
      </AppContextProvider>
    </BrowserRouter>
  );
  const context = wrapper.find('AppContextProvider').instance();
  context.setState({
    lang: 'en'
  });
  expect(wrapper).toMatchSnapshot();
});
