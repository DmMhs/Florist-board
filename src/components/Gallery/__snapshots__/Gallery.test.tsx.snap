// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Gallery>
            <div
              className="Gallery"
            >
              <h2>
                Our Gallery
              </h2>
              <hr />
              <WithStyles(CircularIndeterminate)>
                <CircularIndeterminate
                  classes={
                    Object {
                      "progress": "CircularIndeterminate-progress-1",
                    }
                  }
                >
                  <div
                    className="Spinner"
                  >
                    <WithStyles(CircularProgress)
                      className="CircularIndeterminate-progress-1"
                    >
                      <CircularProgress
                        className="CircularIndeterminate-progress-1"
                        classes={
                          Object {
                            "circle": "MuiCircularProgress-circle-8",
                            "circleDisableShrink": "MuiCircularProgress-circleDisableShrink-11",
                            "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-10",
                            "circleStatic": "MuiCircularProgress-circleStatic-9",
                            "colorPrimary": "MuiCircularProgress-colorPrimary-5",
                            "colorSecondary": "MuiCircularProgress-colorSecondary-6",
                            "indeterminate": "MuiCircularProgress-indeterminate-4",
                            "root": "MuiCircularProgress-root-2",
                            "static": "MuiCircularProgress-static-3",
                            "svg": "MuiCircularProgress-svg-7",
                          }
                        }
                        color="primary"
                        disableShrink={false}
                        size={40}
                        thickness={3.6}
                        value={0}
                        variant="indeterminate"
                      >
                        <div
                          className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5 MuiCircularProgress-indeterminate-4 CircularIndeterminate-progress-1"
                          role="progressbar"
                          style={
                            Object {
                              "height": 40,
                              "width": 40,
                            }
                          }
                        >
                          <svg
                            className="MuiCircularProgress-svg-7"
                            viewBox="22 22 44 44"
                          >
                            <circle
                              className="MuiCircularProgress-circle-8 MuiCircularProgress-circleIndeterminate-10"
                              cx={44}
                              cy={44}
                              fill="none"
                              r={20.2}
                              strokeWidth={3.6}
                              style={Object {}}
                            />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </CircularIndeterminate>
              </WithStyles(CircularIndeterminate)>
            </div>
          </Gallery>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
